umbra-js/classes/RandomNumber.js

```diff
@@ -1,40 +1,53 @@
 /**
- * @notice Class for managing random numbers
+ * @notice Class for managing random numbers. In Umbra, a random number is 32 bytes, where
+ * the first 16 bytes are zeros (optionally used by developers as payload extensions), and the
+ * last 16 bytes are the random number
  */
 const ethers = require('ethers');
-const { padHex } = require('../utils/utils');
-
-const { utils } = ethers;
+const { BigNumber, utils } = ethers;
+const { hexZeroPad } = utils;
+const zeroPrefix = '0x00000000000000000000000000000000'; // 16 bytes of zeros
 
 class RandomNumber {
   /**
    * @notice Generate a new random number
-   * @param {Number} length Number of bytes random number should have
+   * @param {String} payloadExtension 16 byte payload extension, with 0x prefix. Specify as a
+   * 34 character hex string where the first two characters are 0x
    */
-  constructor(length = 32) {
-    this.length = length;
-    this.value = utils.shuffled(utils.randomBytes(length));
-  }
+  constructor(payloadExtension = zeroPrefix) {
+    // Generate default random number without payload extension
+    this.randomLength = 16; // 16 byte random number
+    this.fullLength = 32; // Pad random number to 32 bytes
+    const randomNumberAsBytes = utils.shuffled(utils.randomBytes(this.randomLength));
```

Comment:
Not a change, just wondering, what does the `shuffled` function do here?

Reply:
It just shuffles the random bytes array with a Fisher–Yates shuffle. I figured I'd add that in to get a bit more randomness, but in reality it probably doesn't make much difference

Reply:
Gotcha! I'd say that doesn't hurt, but doesn't really help either. Either the bytes coming out of `randomBytes` are secure randomness as is, or they're not. No need to change, though. Should be good to merge 👍